src/connection.js

```diff
@@ -986,6 +1023,7 @@ class Connection extends EventEmitter {
       }
 
       this.request = request;
+      this.request.connection = this;
```

Comment:
@chdh The implementation is certainly good enough and I'll approve. The couple of tweaks left I can make them after merge. I'll let @arthurschreiber merge the change as he wanted to review before merge.

Thanks for your contribution and thanks for you patience!